assets/js/blocks/cart-checkout/mini-cart-contents/edit.tsx

```diff
@@ -2,16 +2,82 @@
  * External dependencies
  */
 import type { ReactElement } from 'react';
-import { useBlockProps } from '@wordpress/block-editor';
+import {
+	useBlockProps,
+	InnerBlocks,
+	BlockControls,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { Icon, filledCart, removeCart } from '@woocommerce/icons';
+import { EditorProvider } from '@woocommerce/base-context';
+import type { TemplateArray } from '@wordpress/blocks';
 
-const Edit = (): ReactElement => {
+/**
+ * Internal dependencies
+ */
+import { useViewSwitcher, useForcedLayout } from '../shared';
+
+// Array of allowed block names.
+const ALLOWED_BLOCKS = [
+	'woocommerce/filled-mini-cart-contents-block',
+	'woocommerce/empty-mini-cart-contents-block',
+];
+interface Props {
+	clientId: string;
+}
+
+const Edit = ( { clientId }: Props ): ReactElement => {
 	const blockProps = useBlockProps();
 
+	const defaultTemplate = [
+		[ 'woocommerce/filled-mini-cart-contents-block', {}, [] ],
+		[ 'woocommerce/empty-mini-cart-contents-block', {}, [] ],
+	] as TemplateArray;
+
+	const { currentView, component: ViewSwitcherComponent } = useViewSwitcher(
+		clientId,
+		[
+			{
+				view: 'woocommerce/filled-mini-cart-contents-block',
+				label: __( 'Filled Mini Cart', 'woo-gutenberg-products-block' ),
+				icon: <Icon srcElement={ filledCart } />,
+			},
+			{
+				view: 'woocommerce/empty-mini-cart-contents-block',
+				label: __( 'Empty Mini Cart', 'woo-gutenberg-products-block' ),
+				icon: <Icon srcElement={ removeCart } />,
+			},
+		]
+	);
+
+	useForcedLayout( {
+		clientId,
+		registeredBlocks: ALLOWED_BLOCKS,
+		defaultTemplate,
+	} );
+
 	return (
 		<div { ...blockProps }>
-			<p>Editing the mini cart contents</p>
+			<EditorProvider currentView={ currentView }>
+				<BlockControls __experimentalShareWithChildBlocks>
+					{ ViewSwitcherComponent }
+				</BlockControls>
+				<InnerBlocks
+					allowedBlocks={ ALLOWED_BLOCKS }
+					template={ defaultTemplate }
+					templateLock={ false }
+				/>
+			</EditorProvider>
 		</div>
 	);
 };
 
 export default Edit;
+
+export const Save = (): JSX.Element => {
+	return (
+		<div { ...useBlockProps.save() }>
```

Comment:
With the addition of the div, the template no longer matches with the block. (more details on `p1638810187484200?thread_ts=1638803183.465500&cid=C02FL3X7KR6-slack-C02FL3X7KR6`)

Reply:
@gigitux I update the template file, please have another look at this : ).

Reply:
Great! Thanks for the update!